frontend/src/components/common/Filter/components/BasicFilter/index.tsx

```diff
@@ -1,33 +1,49 @@
-import { Popover, Position } from "@blueprintjs/core";
-import { ReactNode } from "react";
-import { Filter } from "../../common/style";
+import { MouseEvent, ReactNode, useState } from "react";
```

Comment:
Refactoring `BasicFilter` component to replace `Blueprint` Popover with `Mui` equivalent. In doing so, we need to restructure `BasicFilter` to take a button (`FilterLabel`) and control open/close action of the popover with state.